### PR TITLE
Lint view components to mark usage of User class methods as offenses

### PR DESCRIPTION
**Pull request by dmarcoux**
This relies on global state and it should be avoided.

### CODE REVIEW

**Comment by dmarcoux** [comment]
This PR should be rebased once #13739 is merged.

**Comment by dmarcoux** [comment]
Fixing all RuboCop offenses found by the new version of `ViewComponent/AvoidGlobalState` would be too much in a single PR, so I decided to regenerate the TODOs. Those can be addressed later in other PRs.

File: src/api/spec/lib/rubocop/cop/view_component/avoid_global_state_spec.rb [hunk 1]
```diff
@@ -15,14 +15,18 @@ def initialize
...
+                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ View components should not rely on global state by calling User.find_by(login: 'someone'). Instead, pass the required data to the initialize method.
 
           end
         end
       RUBY
     end
   end
 
-  context 'when a view component does not use params' do
+  context 'when a view component does not use params or call a User class method' do
```
Comment (eduardoj) [inline src/api/spec/lib/rubocop/cop/view_component/avoid_global_state_spec.rb:29]:
```suggestion
  context 'when a view component does not use params and does not call a User class method' do
```

Comment (dmarcoux) [reply src/api/spec/lib/rubocop/cop/view_component/avoid_global_state_spec.rb:29]: The meaning is the same, but if you prefer this version, no worries!

File: src/api/spec/lib/rubocop/cop/view_component/avoid_global_state_spec.rb [hunk 2]
```diff
@@ -34,11 +38,12 @@ def initialize
     end
   end
 
-  context 'when a class which is not a view component uses params' do
+  context 'when a class which is not a view component uses params or call a User class method' do
```
Comment (eduardoj) [inline src/api/spec/lib/rubocop/cop/view_component/avoid_global_state_spec.rb:41]:
```suggestion
  context 'when a class which is not a view component uses params or calls a User class method' do
```

**Review by eduardoj — APPROVED** [review]

**Comment by dmarcoux** [comment]
The failing check `OBS: obs-server:obs-bundled-gems - 15.4/x86_64` is completely unrelated to the changes in this PR, it's about a rack version mismatch.

**Pull request merged**